### PR TITLE
Augmentation methods

### PR DESCRIPTION
**Pull request by AlexEMG**
Changed **default augmentation method to imgaug.** 

- this consistently gives more robust results and has many more possiblities (see https://github.com/DeepLabCut/DeepLabCut/wiki/Data-Augmentation and https://github.com/DeepLabCut/DeepLabCut/blob/master/docs/functionDetails.md#f-create-training-datasets)
- since the _previous default_ did not have a descriptive name, it is now called "scalecrop". This is the method introduced in Nath et al. https://www.nature.com/articles/s41596-019-0176-0

Ran all testscripts & quantitatively on augmentation comparison got:
***Default = Imagaug**** augmentation:

Results for 20000  training iterations: 95 1 train error: 3.35 pixels. Test error: 3.64  pixels.
With pcutoff of 0.4  train error: 3.35 pixels. Test error: 3.64 pixels

Scalecrop (was = default) augmentation:

Results for 20000  training iterations: 95 2 train error: 2.69 pixels. Test error: 3.28  pixels.
With pcutoff of 0.4  train error: 2.69 pixels. Test error: 3.28 pixels

Tensorpack augmentation:

Results for 20000  training iterations: 95 3 train error: 3.39 pixels. Test error: 4.26  pixels.
With pcutoff of 0.4  train error: 3.39 pixels. Test error: 4.26 pixels

Notice: despite the higher RMSE for imgaug due to the augmentation,
the network performs much better on the testvideo.

### CODE REVIEW

**Comment by AlexEMG** [comment]
Note, some of the changes you see (in e.g. labeling toolbox)  are due to black.

**Review by MMathisLab — COMMENTED** [review]
I would like to see defaults for imgaug updated, as in here: https://github.com/DeepLabCut/DeepLabCut/wiki/Data-Augmentation
cfg_dlc['elastic_transform']=True
cfg_dlc['rotation']=180
cfg_dlc['covering']=True
cfg_dlc['motion_blur'] = True

**Comment by AlexEMG** [comment]
Add an angle range variable to the default pose_cfg.yaml
Rerun testscript

**Comment by AlexEMG** [comment]
https://imgaug.readthedocs.io/en/latest/source/overview/color.html

Put example how to add additional ones.

**Comment by AlexEMG** [comment]
My results were (Run with DLC *2.2b7* on August 1st 2020) for 10k iterations

Imgaug:
Results for 10000  training iterations: 95 1 train error: 3.78 pixels. Test error: 3.89  pixels.
With pcutoff of 0.4  train error: 3.78 pixels. Test error: 3.89 pixels

Scalecrop:
Done and results stored for snapshot:  snapshot-10000
Results for 10000  training iterations: 95 2 train error: 2.81 pixels. Test error: 2.46  pixels.
With pcutoff of 0.4  train error: 2.81 pixels. Test error: 2.46 pixels

Tensorpack:
Done and results stored for snapshot:  snapshot-10000
Results for 10000  training iterations: 95 3 train error: 3.76 pixels. Test error: 3.98  pixels.
With pcutoff of 0.4  train error: 3.76 pixels. Test error: 3.98 pixels

But the testvideo looks perfect for either Tensorpack and Imgaug, but has a flip for scalecrop.

File: deeplabcut/generate_training_dataset/trainingsetmanipulation.py [hunk 1]
```diff
@@ -533,13 +533,21 @@ def ParseYaml(configfile):
     return docs
 
 
-def MakeTrain_pose_yaml(itemstochange, saveasconfigfile, defaultconfigfile):
+def MakeTrain_pose_yaml(
+    itemstochange, saveasconfigfile, defaultconfigfile, items2drop={}
+):
     docs = ParseYaml(defaultconfigfile)
+    for key in items2drop.keys():
+        print(key, "dropping?")
```
Comment (jeylau) [inline deeplabcut/generate_training_dataset/trainingsetmanipulation.py]: Do we need that print statement here?

**Review by jeylau — APPROVED** [review]

File: deeplabcut/generate_training_dataset/trainingsetmanipulation.py (hunk 1 above)
Comment (AlexEMG) [reply deeplabcut/generate_training_dataset/trainingsetmanipulation.py]: Nope, removed :)

**Pull request merged**